rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/parsetreetoast/data_definitions.kt

```diff
@@ -400,12 +400,16 @@ fun RpgParser.Parm_fixedContext.toTypeInfo(): TypeInfo {
 
 internal fun RpgParser.Parm_fixedContext.calculateExplicitElementType(arraySizeDeclared: Int?): Type? {
     val rpgCodeType = DATA_TYPE()?.text?.trim() ?: RpgType.ZONED.rpgType
-    val precision = if (TO_POSITION().text.isNotBlank()) TO_POSITION().text.trim().toInt() else null
     val decimalPositions = if (DECIMAL_POSITIONS().text.isNotBlank()) with(DECIMAL_POSITIONS().text.trim()) { if (isEmpty()) 0 else toInt() } else null
     val isPackEven = keyword().any { it.keyword_packeven() != null }
     val isVarying = keyword().any { it.keyword_varying() != null }
     val startPosition = this.explicitStartOffset()
     val endPosition = this.explicitEndOffset()
+    val precision = if (startPosition != null && endPosition != null) {
```

Comment:
Si potrebbe anche pensare di spostarlo in un extension method separato

Reply:
Scusa, ma non penso di aver capito il tuo suggerimento. Di che tipo vorresti fare l'estensione?

Reply:
Qualcosa del tipo:
```
fun RpgParser.Parm_fixedContext.precision() =  if (startPosition != null && endPosition != null) {endPosition - startPosition.toInt()
    } else {
        if (TO_POSITION().text.isNotBlank()) TO_POSITION().text.trim().toInt() else null
    }
```

O anche:

```
val RpgParser.Parm_fixedContext.precision
get() =  if (startPosition != null && endPosition != null) {endPosition - startPosition.toInt()
    } else {
        if (TO_POSITION().text.isNotBlank()) TO_POSITION().text.trim().toInt() else null
    }
```

Reply:
startPosition ed endPosition non sono attributi di RpgParser.Parm_fixedContex, ma variabili calcolate in calculateExplicitElementType ed usate in diversi punti della funzione. Ci devo pensare